dfdaemon/config/config.go

```diff
@@ -413,3 +417,14 @@ func NewProxy(regx string, useHTTPS bool, direct bool, redirect string) (*Proxy,
 func (r *Proxy) Match(url string) bool {
 	return r.Regx != nil && r.Regx.MatchString(url)
 }
+
+// ProtocolConfig is the config of url protocol.
+type ProtocolConfig struct {
+	Name string `yaml:"name" json:"name"`
+	// Opts defines the opts for protocol, such as tls config.
+	Opts map[string]string `yaml:"opts" json:"opts"`
```

Comment:
If this config will be available for plugins, maybe we should use `map[string]interface{}` to allow more complicated configuration.